internal/install/types/errors.go

```diff
@@ -2,24 +2,65 @@ package types
 
 import (
 	"errors"
+	"fmt"
+	"regexp"
 )
 
 // ErrInterrupt represents a context cancellation.
 var ErrInterrupt = errors.New("operation canceled")
 
-// ErrNonZeroExitCode represents a non-zero exit code bieing returned from a child process.
+type GoTaskError interface {
+	error
+	Tasks() []string
+}
+
+// GoTaskError represents a task failure reported by go-task.
+type GoTaskGeneralError struct {
+	error
+	tasks []string
+}
+
+func (e GoTaskGeneralError) Error() string {
+	return e.error.Error()
+}
+
+func (e GoTaskGeneralError) Tasks() []string {
+	return e.tasks
+}
+
+func NewGoTaskGeneralError(err error) GoTaskError {
+	re := regexp.MustCompile(`task: Failed to run task \"default\": task: Failed to run task \"(.+)\": `)
+
+	parsed := re.FindAllSubmatch([]byte(err.Error()), 1)
+
+	var task string
+	if len(parsed) > 0 && len(parsed[0]) > 0 {
+		task = string(parsed[0][1])
+	}
+
+	stripped := re.ReplaceAllString(err.Error(), "")
+
+	return GoTaskGeneralError{
+		tasks: []string{task},
```

Comment:
Does 'tasks' hold more than 1 task?  It looks like the code is setup to find 1 and only 1 task.  Maybe we should call it just 'task'?

Reply:
This involves a change to the contract between the front and back end. I wanted to leave the type open to defining a variable length path to support arbitrary task nesting, but didn't implement just yet.